CDP4Composition/CommonView/ViewModels/AttachmentViewModel.cs

```diff
@@ -0,0 +1,650 @@
+        /// Remove a <see cref="FileType"/> from the list of <see cref="FileType"/>s
+        /// </summary>
+        private void DeleteFileType()
+        {
+            this.FileType.Remove(this.SelectedFileType);
+        }
+
+        /// <summary>
+        /// Add the file of a physical file to this <see cref="Attachment"/>
+        /// </summary>
+        private void AddFile()
+        {
+            var result = this.fileDialogService.GetOpenFileDialog(false, false, false, string.Empty, string.Empty, string.Empty, 1);
+
+            if (result.Length != 1)
+            {
+                return;
+            }
+
+            var currentPath = result.First();
+            var fileName = System.IO.Path.GetFileName(currentPath);
+
+            this.LocalPath = currentPath;
+            this.FileType.Clear();
+
+            if (fileName != null)
+            {
+                var extensionArray = fileName.Split(new[] { "." }, StringSplitOptions.None);
+                var fileTypes = new List<FileType>();
+
+                for (var i = extensionArray.Length - 1; i >= 0; i--)
+                {
+                    var fileType = this.PossibleFileType.FirstOrDefault(x => x.Extension.ToLower().Equals(extensionArray[i].ToLower()));
+
+                    if (fileType == null)
+                    {
+                        break;
+                    }
+
+                    fileTypes.Insert(0, fileType);
+                    fileName = string.Join(".", extensionArray.Take(i));
+                }
+
+                this.FileType.AddRange(fileTypes);
+            }
+
+            this.Name = fileName;
+        }
+
+        /// <summary>
+        /// Add a <see cref="FileType"/> to the list of <see cref="FileType"/>s
+        /// </summary>
+        private void AddFileType()
+        {
+            var result = this.thingSelectorDialogService.SelectThing(
+                this.PossibleFileType.Except(this.FileType).ToList(),
+                new List<string> { "Name", "ShortName", "Extension" });
+
+            if (result is not null && !this.FileType.Contains(result))
+            {
+                this.FileType.Add(result);
+            }
+        }
+
+        /// <summary>
+        /// Actions to do after FileType changes
+        /// </summary>
+        private void AfterUpdateFileType()
+        {
+            this.CanMoveUpFileType = !this.IsReadOnly &&
+                                     this.SelectedFileType is not null &&
+                                     this.FileType.Count > 1 &&
+                                     this.FileType.IndexOf(this.SelectedFileType) > 0;
+
+            this.CanMoveDownFileType = !this.IsReadOnly &&
+                                       this.SelectedFileType is not null &&
+                                       this.FileType.Count > 1 &&
+                                       this.FileType.IndexOf(this.SelectedFileType) < this.FileType.Count - 1;
+
+            this.UpdatePath();
+        }
+
```

Comment:
This file has a few places with consecutive empty lines, can you clean it up please?